gateways/stripe/espresso_stripe.js

```diff
@@ -0,0 +1,139 @@
+		this.cardHolderCountry = {};
+		this.cardHolderZip = {};
+		this.initialized = false;
+
+		/**
+		 * @function initialize
+		 */
+		this.initialize = function() {
+			this.initialize_objects();
+		
+			// ensure that the Stripe js class is loaded
+			if ( typeof Stripe === 'undefined' ) {
+				return;
+			}
+
+			this.stripe = Stripe( ee_stripe_args.stripe_pk_key );
+			const args = {};
+			if ( ee_stripe_args.data_locale ) {
+				args.locale = ee_stripe_args.data_locale;
+			}
+			this.stripeElements = this.stripe.elements( args );
+			this.stripeCardElement = this.stripeElements.create( 'card' );
+			this.stripeCardElement.mount( '#ee-stripe-card-element' );
+			
+			// Set up listener for payment button
+			this.set_listener_for_submit_payment_button();
+
+			//alert('this.initialized');
+			this.initialized = true;
+		};
+
+		/**
+		 * @function initialize_objects
+		 */
+		this.initialize_objects = function() {
+			//console.log( JSON.stringify( '**this.initialize_objects**', null, 4 ) );
+			this.submitPaymentButton = $( this.submitButtonId );
+			this.paymentIntentInput = $( '#espresso_stripe_payment_intent_id' );
+			this.stripeErrorContainer = $( '#espresso_stripe_errors' );
+			this.cardHolderFirstName = $( '#espresso_stripe_first_name' );
+			this.cardHolderLastName = $( '#espresso_stripe_last_name' );
+			this.cardHolderEmail = $( '#espresso_stripe_email' );
+			this.cardHolderAddress = $( '#espresso_stripe_address' );
+			this.cardHolderCity = $( '#espresso_stripe_city' );
+			this.cardHolderState = $( '#espresso_stripe_state' );
+			this.cardHolderCountry = $( '#espresso_stripe_country' );
+		};
+
+		/**
+		 * @function set_listener_for_submit_payment_button
+		 */
+		this.set_listener_for_submit_payment_button = function() {
+			const stipe_instance = this;
+			//console.log( 'initialize', 'set_listener_for_submit_payment_button', true );
+			this.submitPaymentButton.on( 'click', function(e) {
+				e.preventDefault();
+				const billingDetails = {};
+				if ( stipe_instance.cardHolderFirstName.val() || stipe_instance.cardHolderLastName.val() ) {
+					billingDetails.name = stipe_instance.cardHolderFirstName.val() + ' ' + stipe_instance.cardHolderLastName.val();
+				}
+				if ( stipe_instance.cardHolderEmail.val() ) {
+					billingDetails.email = stipe_instance.cardHolderEmail.val();
+				}
+
+				const address = {};
+				if ( stipe_instance.cardHolderAddress.val() ) {
+					address.line1 = stipe_instance.cardHolderAddress.val();
+				}
+				if ( stipe_instance.cardHolderCity.val() ) {
+					address.city = stipe_instance.cardHolderCity.val();
+				}
+				if ( stipe_instance.cardHolderState.val() ) {
+					address.state = stipe_instance.cardHolderState.val();
+				}
+				if ( stipe_instance.cardHolderCountry.val() ) {
+					address.country = stipe_instance.cardHolderCountry.val();
+				}
+				if ( Object.keys( address ).length > 0 ) {
+					billingDetails.address = address;
+				}
+				stipe_instance.stripe.handleCardPayment( 
```

Comment:
I think using `handleCardPayment` turns out to be fine because you're doing the "[Automatic confirmation](https://stripe.com/docs/payments/payment-intents/web)" approach; whereas EE4 uses "[Manual confirmation](https://stripe.com/docs/payments/payment-intents/web-manual)".
I originally thought Automatic Confirmation required an IPN to the server, but it looks like the message they send is just to the JS, which sounds perfectly acceptable.
It sounds like Automatic Confirmation has you: 
1. Server side: Create the payment intent
2. Client side: Use that, and the billing form, to charge card and store the result
3. Server side: Update the payment in our system :ok_hand: 

Manual Confirmation has you:
1. Client side: Create a Stripe payment method and store its ID
2. Server side: Create a payment intent, and use the payment method's ID to charge the card UNLESS it requires 3D secure, in which case...
3. Client side: We do 3D secure, and then store the 3D-authed payment ID
4. Server side: Attempt to complete the payment server-side

I think Automatic confirmation sounds fine provided the payment amount won't change once we show them the billing form (or we listen for when the payment amount changes, and then create a new payment intent server-side, and then update the client-side payment intent ID variable.) I don't think EE3 needs to worry about the payment amount changing once the billing form is shown right? (Whereas for EE4 it was more of an issue). So Automatic Confirmation (`handleCardPayment`) may be the better choice for EE3, whereas Manual Confirmation (`handleCardAction`) may still be better for EE4. :smiley: